include/Exceptions.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <exception>
+#include <string>
+
+namespace hemiola
+{
+    /*!
+     * @brief Exception to be used when something specific to KeyboardEvents should throw.
+     */
+    class KeyboardException : public std::exception
+    {
+    public:
+        explicit KeyboardException ( const std::string& msg )
+            : m_Msg { msg }
+        {}
+
+        virtual const char* what() const noexcept { return m_Msg.c_str(); }
+
+    private:
+        /*!
+         * @brief string associated with the exception
+         */
+        std::string m_Msg;
+    };
+
+    /*!
+     * @brief An exception that has an event code associated with it.
```

Comment:
Minor: be a little more specific about the type of code in the comment.  That way it's not ambiguous if it's something from a library, an errno, some other enum etc.